### ISSUE TITLE
always `state changed` for recursive `wildfly::util::resource`

### ISSUE THREAD

**Issue by bjwschaap** [biemond/biemond-wildfly#108]
I have a recursive resource. E.g. (partial example):
```
wildfly::util::resource { "/core-service=management/security-realm=${realm_name}/authorization=ldap":
    recursive => true,
    content   => {
      'connection' => "${realm_name}-LDAPConnection",
      'group-search' => {
        'group-to-principal' => {
          'group-name'                 => $authorization_group_name,
          'group-name-attribute'       => $authorization_group_name_attribute,
          'group-dn-attribute'         => $authorization_group_dn_attribute,
          'base-dn'                    => $authorization_group_base_dn,
          'search-by'                  => $authorization_group_search_by,
          'principal-attribute'        => $authorization_principal_attribute,
          'recursive'                  => $authorization_group_recursive,
          'iterative'                  => $authorization_group_iterative,
          'prefer-original-connection' => $authorization_prefer_original_conn,
          'cache' => {
            "${cache_type}" => {
              'max-cache-size' => $max_cache_size,
              'eviction-time'  => $cache_eviction_time,
              'cache-failures' => $cache_failures,
            }
          }
      }},
      'username-to-dn' => {
        'username-filter' => {
          'base-dn'           => $ldap_user_base_dn,
          'attribute'         => $authorization_user_name_attribute,
          'user-dn-attribute' => $authorization_user_dn_attribute,
          'force'             => $authorization_user_force,
          'recursive'         => $authorization_user_recursive,
          'cache' => {
            "${cache_type}" => {
              'max-cache-size' => $max_cache_size,
              'eviction-time'  => $cache_eviction_time,
              'cache-failures' => $cache_failures,
            }
          }
      }},
    },
  }
```
Each Puppet run will cause a `state changed` because Puppet thinks the resources are dissimilar. This is incorrect since the resources **are similar**.

**Comment by bjwschaap** [comment]
I observed that the `insync?` method in `lib/puppet/type/wildfly_resource.rb` doesn't sort nested hashes correctly:

should:
```
["[\"connection\", \"MyOrgLDAPRealm-LDAPConnection\"]", "[\"group-search\", {\"group-to-principal\"=>{\"group-name\"=>\"SIMPLE\", \"group-name-attribute\"=>\"cn\", \"group-dn-attribute\"=>\"dn\", \"base-dn\"=>\"cn=gebruikersgroepen,ou=diensten,o=MyOrg,c=nl\", \"search-by\"=>\"DISTINGUISHED_NAME\", \"principal-attribute\"=>\"member\", \"recursive\"=>true, \"iterative\"=>false, \"prefer-original-connection\"=>true, \"cache\"=>{\"by-access-time\"=>{\"max-cache-size\"=>\"1000\", \"eviction-time\"=>\"900\", \"cache-failures\"=>false}}}}]", "[\"username-to-dn\", {\"username-filter\"=>{\"base-dn\"=>\"cn=users,ou=diensten,o=MyOrg,c=nl\", \"attribute\"=>\"uid\", \"user-dn-attribute\"=>\"dn\", \"force\"=>false, \"recursive\"=>false, \"cache\"=>{\"by-access-time\"=>{\"max-cache-size\"=>\"1000\", \"eviction-time\"=>\"900\", \"cache-failures\"=>false}}}}]"]
```
Is:
```
["[\"connection\", \"MyOrgLDAPRealm-LDAPConnection\"]", "[\"group-search\", {\"group-to-principal\"=>{\"base-dn\"=>\"cn=gebruikersgroepen,ou=diensten,o=MyOrg,c=nl\", \"group-dn-attribute\"=>\"dn\", \"group-name\"=>\"SIMPLE\", \"group-name-attribute\"=>\"cn\", \"iterative\"=>false, \"prefer-original-connection\"=>true, \"principal-attribute\"=>\"member\", \"recursive\"=>true, \"search-by\"=>\"DISTINGUISHED_NAME\", \"cache\"=>{\"by-access-time\"=>{\"cache-failures\"=>false, \"eviction-time\"=>900, \"max-cache-size\"=>1000}}}}]", "[\"username-to-dn\", {\"username-filter\"=>{\"attribute\"=>\"uid\", \"base-dn\"=>\"cn=users,ou=diensten,o=MyOrg,c=nl\", \"force\"=>false, \"recursive\"=>false, \"user-dn-attribute\"=>\"dn\", \"cache\"=>{\"by-access-time\"=>{\"cache-failures\"=>false, \"eviction-time\"=>900, \"max-cache-size\"=>1000}}}}]"]
```

**Comment by bjwschaap** [comment]
The 'fix' for issue #97 breaks recursive resources....

**Comment by bjwschaap** [comment]
Thanks @jairojunior !
